views/CreatePost/CreatePost.jsx

```diff
@@ -0,0 +1,423 @@
+          navigation.goBack();
+        }
+      })
+      .catch(err => {
+        console.error(err);
+        alert("Error updating post. Sorry about that!");
+      });
+  }
+
+  textUpdate = (text) => {
+    this.setState({ resourceText: text })
+  }
+
+  takeImage = async() => {
+    const { status: cameraPerm } = await Permissions.getAsync(
+      Permissions.CAMERA
+    );
+    let cameraStatus = cameraPerm;
+
+    if (cameraPerm !== 'granted') {
+      const { status } = await Permissions.askAsync(Permissions.CAMERA);
+      cameraStatus = status;
+    }
+
+    const { status: cameraRollPerm } = await Permissions.getAsync(
+      Permissions.CAMERA_ROLL
+    );
+    let cameraRollStatus = cameraRollPerm;
+
+    if (cameraRollPerm !== 'granted') {
+      const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
+      cameraRollStatus = status;
+    }
+
+    if (cameraRollStatus !== 'granted' || cameraStatus !== 'granted') {
+      return null;
+    }
+
+    let result = await ImagePicker.launchCameraAsync({
+      allowsEditing: true,
+      base64: true,
+      quality: 0.5,
+    });
+    this.setState({
+      image: result.uri,
+      imageUpdated: true,
+    });
+  }
+
+  pickImage = async () => {
+    if (Platform.OS === 'ios') {
+      const { status: existingStatus } = await Permissions.getAsync(
+        Permissions.CAMERA_ROLL
+      );
+      let finalStatus = existingStatus;
+
+      if (existingStatus !== 'granted') {
+        const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
+        finalStatus = status;
+      }
+
+      if (finalStatus !== 'granted') {
+        return null;
+      }
+    }
+
+    let result = await ImagePicker.launchImageLibraryAsync({
+      allowsEditing: true,
+      base64: true,
+      quality: 0.5,
+    });
+    this.setState({
+      image: result.uri,
+      imageUpdated: true,
+    });
+  }
+
+  addPoll = async () => {
+    this.setState({
+      isPoll: true,
+      image: null,
+      pollUpdated: true
+    });
+  }
+
+  clearMedia = async () => {
+    this.setState({
+      isPoll: false,
+      image: null,
+      imageUpdated: !!this.state.image,
+      pollUpdated: this.state.isPoll
+    });
+  }
+
+  updatePoll = async (data) => {
+    this.setState({
+      isPoll: !!data,
+      pollData: data,
+      pollUpdated: true
+    });
+  }
+
+  updateChannel = (value) => {
+    this.setState({
+      selectedChannel: value
+    })
+  }
+
+  render() {
+    const displayToolbar = !this.state.isPoll && !this.state.image
+
+    var postPlaceholder = "Write, post, recieve community..."
+    if(this.state.isPoll){
+      postPlaceholder = "What questions do you have?"
+    }
+
+    return (
+      <Container>
+      { /* Hack to dismiss keyboard when text isn't focused */}
+      <KeyboardAwareScrollView keyboardShouldPersistTaps='handled' scrollEnabled={false}>
+          <View style={styles.selectChannelView}>
+            <RNPickerSelect
```

Comment:
Something seems off with this picker. Not sure how it works, but it seems like sometimes it doesn't register my taps. Sometimes I have to tap 5 times for it to work.

Reply:
It's an Android bug... do you feel like it's blocking this PR? I know it has to be fixed, but I wasn't terrible on my Android device. It might take a few hours to fix

Reply:
Meh, fine for now - because I do think it's an improvement over the current post flow. Definitely something to not forget about though.